lang/gcc10-bootstrap/Portfile

```diff
@@ -0,0 +1,250 @@
+                    mpc-1.2.1.tar.gz \
+                    rmd160  8c25b71d080936aab6ffa84f4f9d99b662d82c47 \
+                    sha256  17503d2c395dfcf106b622dc142683c1199431d095367c6aacba6eec30340459 \
+                    size    838731 \
+                    isl-0.24.tar.bz2 \
+                    rmd160  b3bf8e1ad50207d4eebecc47cb4cffdac6581a57 \
+                    sha256  fcf78dd9656c10eb8cf9fbd5f59a0b6b01386205fe1934b3b287a0a1898145c0 \
+                    size    2261594
+
+extract {
+    foreach f ${distfiles} {
+        set f [lindex [split ${f} :] 0]
+        ui_msg "Extracting: ${f}"
+        if {[string match "*bz2*" ${f}]} {
+            ui_msg "${workpath} \"/usr/bin/tar xjf ${distpath}/${f}\""
+            system -W ${workpath} "/usr/bin/tar xjf ${distpath}/${f}"
+        }
+        if {[string match "*gz*" ${f}]}  {
+            ui_msg "${workpath} \"/usr/bin/tar xzf ${distpath}/${f}\""
+            system -W ${workpath} "/usr/bin/tar xzf ${distpath}/${f}"
+        }
+    }
+}
+
+post-extract {
+    ln -s ${workpath}/${MPFR} ${worksrcpath}/mpfr
+    ln -s ${workpath}/${GMP}  ${worksrcpath}/gmp
+    ln -s ${workpath}/${MPC}  ${worksrcpath}/mpc
+    ln -s ${workpath}/${ISL}  ${worksrcpath}/isl
+}
+
+patchfiles  patch-dynamic-lookup-11.diff
+
+# Branches from the Darwin maintainer of GCC
+if { ${os.arch} eq "arm" } {
+    # https://github.com/iains/gcc-10-branch/tree/gcc-10-3-apple-si
+    patchfiles-append patch-iains-apple-si.diff
+} elseif { ${os.arch} eq "powerpc" } {
+    # https://github.com/iains/gcc-10-branch/tree/gcc-10-3-ppc
+    patchfiles-append patch-iains-ppc.diff
+    # https://gcc.gnu.org/git/gitweb.cgi?p=gcc.git;h=4c3792d448964f7bd99e7eac2c29c9eb7c2bfb84
+    patchfiles-append patch-gcc10-darwin-ppc64-invalid-non-pic.diff
+} elseif { ${os.platform} eq "darwin" } {
+    # https://github.com/iains/gcc-10-branch/tree/gcc-10-3-darwin
+    patchfiles-append patch-iains-darwin.diff
+}
+
+if { ${os.platform} eq "darwin" && ${os.major} > 20} {
+    # Add preliminary Darwin 21 support to GCC 11.
+    # This should be removed, when GCC 11 supports Darwin 21 officially and the
+    # respective version is used by macports.
+    # See also: https://github.com/iains/gcc-darwin-arm64/commit/20f61faaed3b335d792e38892d826054d2ac9f15
+    patchfiles-append patch-darwin21-support.diff
+
+    # This following patch works around a problem on MacOS 12.0 with fortran
+    # See also: https://gcc.gnu.org/git/?p=gcc.git;a=patch;h=fabe8cc41e9b01913e2016861237d1d99d7567bf
+    patchfiles-append patch-darwin21-fortran.diff
+
+    # Additional patch that replace all darwin20* to darwin2*
+    patchfiles-append patch-darwin21-global-replace.diff
+}
+
+if { ${os.platform} eq "darwin" && ${os.major} > 19} {
+    # replace std=gnu++98 to std=c++ on macOS 11 and 12 fixes linking error
+    # See: https://trac.macports.org/ticket/64316
+    patchfiles-append patch-darwin21-cxx11.diff
+}
+
+if { ${configure.build_arch} eq "i386" } {
+
```

Comment:
these fixes should not be needed for this gcc10-bootstrap port, as it will always build (hopefully) with the system compiler and the system cctools/ld64. Having said that, having them here doesn't seem likely to cause any troubles, and might in the end prevent a few tickets if somebody does something wonky somehow and manages to pull in MacPorts' toolchain to build this instead of the system toolchain. 

That might cause some other linkage mess I can't think of yet, but hey -- we like tickets :>

Reply:
@kencu thanks. I've tried to figured out did I need it or not, and just kept it stay on safe side.

Removed.